### PR TITLE
Reduces View Range

### PR DESCRIPTION
**Pull request by TheDZD**
Reduces view range back to 15x15. This has been discussed a little bit recently. Reverts the view range to 15x15, it was changed when UI buttons were originally added as they reduced the amount of space available on screen. After UI buttons were refactored, players gained the ability to minimize UI buttons, meaning most of the screen would be available if needed.

I'm not particularly convinced one way or the other as to whether the size should remain 17x17 or not. Players will need a period of time to adjust to the reduced view range as they had to adjust to the increased one. This should also fix the problem where sprites are distorted by the 'stretch to fit' option.

### CODE REVIEW

**Comment by tigercat2000** [comment]
:-1: I don't see any reason for this at all. We just got our UI button overhaul allowing for so many more UI buttons, and this would just limit that. Distorted sprites are hardly a big problem, if you ask me.

**Comment by ZomgPonies** [comment]
Advantage: MAYBE a little less client lag due to smaller amount of tiles needing to be shown.

Disadvantage: Fucks with the submissions in our sprite contests, will require huge amounts of work to fix all the UI, projectile stuff (the screen size is used somewhere in there), etc.

After playing other servers who run 15x15 to compare I'll have to :-1:, I like our screen space better.

**Comment by Robertha89** [comment]
:-1: this is one of the unique features that paradise has, and much has been made to adapt to that new size.,

**Comment by Fox-McCloud** [comment]
Definitely agree with this. While there's a lot of elements I like about 17x17, on purely selfish basis--overall, the objective fact remains: everything in SS13 is built around, designed for, and meant to be experienced with 15x15.

Voice range, sound range, projectile range, cinematic animations, and proper sprite sizes (tired of having the right side of my head smashed)---easier to click on items that require higher amounts of precision, etc.

Also, with the new lighting system, this means it will help reduce client size lag (and potentially crashes) as it's not forcing the client to load (66,000 less pixels to load!)

With the new action button moveable UI, there's not much of a point to having it be 17x17 anymore.


It resolves a lot of balance issues, completely fixes up stretched sprites and makes things far more pleasant to look at.

**Comment by Flip-Sandwich** [comment]
Eh. People on 1900x1080ish monitors will be able to use a proper 64x64 flat double size instead of terrible mangling from stretch to fit.

That's about it.

**Comment by ZomgPonies** [comment]
DZD - "This should also fix the problem where sprites are distorted by the 'stretch to fit' option." You should specify that fixes the issue for people running at 1900x1080.

**Comment by KasparoVy** [comment]
WAIT WHAT
I'll be able to use 64x64 now!?!? Holy moly

**Comment by fj45** [comment]
No to this. 

Having the extra bit of view range is great. In all my time playing, I have not seen issues with it. (Apart from the stretch to fit thing but thats minor.)

**Comment by fj45** [comment]
From the IRC:

<Mijhead> I like 17x17.
<Mijhead> In some departements, its the difference between being isolated and being able to see people walking by

**Comment by KasparoVy** [comment]
Any pics where that's exemplified? In science the only way you'd be able to miss people walking by is if you were in the very back of the room or in the inner hall looking out into the main hall b/w dept. and the Chapel (just an arbitrary example)

**Comment by Fox-McCloud** [comment]
@KasparoVy 

I'll get some soonish.

**Comment by Fox-McCloud** [comment]
ok, here's some pics

Objective differences:



How it appears more to players:

**Comment by TheDZD** [comment]
> DZD - "This should also fix the problem where sprites are distorted by the 'stretch to fit' option." You should specify that fixes the issue for people running at 1900x1080.
> - ZomgPonies

Wasn't aware of this fact. I know that a lot of the people who complained about this issue use 1080p monitors, but I myself do not (still a 16:9 resolution, but I believe it's 1600x900). I am also not very observant when it comes to graphical issues, so it may be that I simply fail to notice to the stretching.

**Comment by SamCroswell** [comment]
:+1: 
Especially with Meta being ported by Tigercat ( I believe ), I think it's a good idea to be on-par with Baystation and /TG/'s view range. There are balance implications ( Admittedly, not huge ones ) as well as map design ones.

A notable change will come in combat, I suspect. Targets will be bigger, and thus easier to hit for both sides. That said, with a more limited range of view sneaking around might also become easier.

The view range was changed to make room for more buttons on the hotbar, as it was expanded. Now that players can dynamically change the position of the buttons on the hotbar at will, I think the view range should be reverted.

**Comment by KasparoVy** [comment]
That's not so bad. I'd love it for the effect it has on the sprites and 64x64 :D.

Nice touch putting a Vox in there. :+1

**Comment by ZomgPonies** [comment]
I don't mind either view range after all, I was more concerned people were putting an emphasis on balance when the implications were really quite minor.

**Comment by tigercat2000** [comment]
I still can't say I support this. I like the view range as it is, I dislike most aspects of 15,15, and my opinion isn't going to change any time soon.

**Comment by Spacemanspark** [comment]
:-1: I prefer it the way it has been. Please no.

**Comment by Earthdivine** [comment]
I would thumbs up this if ponies didn't bring up such good points. A lot of our stuff would need to be fixed to this new view range and our lobby contest would be a mess to fix, since the screens people are making are now too big for the lobby. 

This feels like it'll create a far bigger mess then actually fixing anything other then making things pretty. The combat bonuses would be very minor as it's not clicking the sprite that's the problem, but clicking it and having it register through lag. 

Tl;dr, it would be a thumbs up if it didn't create so many problems, so I'm against this

**Comment by SamCroswell** [comment]
The screens people are making for the lobbies can just be scaled down, no? All of the submissions I'm seeing aren't already in the .dmi format anyhow, which makes it all the easier.

**Comment by Fox-McCloud** [comment]
@Earthdivine 

It actually won't; most of what's in the codebase is already adjusted for 15x15; it's really only a few UI elements and a single file that needs to be replaced to get us back in line with this; cinematics, some vision stuff, projectiles, etc.

It would actually be 10x more work to get everything in line with 17x17 than to go back to 15x15, believe it or not.

As for the contest; it's as simple as taking the winning photo and shrinking it down, a bit, with Photoshop or some other photo editing software of choice; typicallly, with a lot of contests, you usually end up sending a *high resolution* photo and they make it *lower resolution* at the end of the day anyway.

**Comment by ZomgPonies** [comment]
"It would actually be 10x more work to get everything in line with 17x17 than to go back to 15x15, believe it or not."

Really not confident about that.

**Comment by Krausus** [comment]
So, this would take away ~22% of what players can currently see, forcing everyone to get re-accustomed to a smaller view, requiring UI elements to be shuffled around, and causing who knows what issues with things that have been adjusted for 17x17...

... so icons won't be a little distorted at certain resolutions, people can't see a tile further than some things are "balanced" for, porting some things is easier, and so we'll conform to how SS13 is "meant to be experienced"?

None of that seems worth the impact, albeit temporary, this will have on players. :-1:

**Comment by Dinarzad** [comment]
Personal perspective is I'd prefer the 15x15. It's easier on my eyes, it's less lag, less to load at one time.

I don't personally believe the added vision range adds enough to the game.
's my opinion on it anyway.
So.
Yea.
Thumbs are up for it.

**Comment by Earthdivine** [comment]
If the other modifications can be done in this PR, the changes to the UI and fixing what ever is needed with guns, I would be ok with merging. But right now I don't want it to be "it'll be easy to fix" and jut forget about it. Something like this needs to get done fully at once or we run the risk of forgetfulness.

**Comment by ZomgPonies** [comment]
In favor: 1 maintainer, 1 seniormin and two randoms
Against: 1 maintainer, 2 staff coders, and 3 randoms
On the Fence: 1 maintainer, 1 headmin (against until the changes are fully implemented and tested)

If you can show empirical proof that these changes will help with lag/client crashing, that would probably help this PR pass, but until that's done I don't think this PR will garner enough support.

**Comment by Plotron** [comment]
Limit the view range to claustrophobia-inducing 15x15? 17x17 being the one special thing about Paradise that makes me play it instead of other servers? Well, I'm certainly biased, but I can't see how clearer image and minor sprite fixes are worth the switch.

IMO 15x15 is too tiny. Cyberiad has a lot of big, open rooms, where view range counts.
If you want to make SS13 look even more archaic, then sure, switch back to 15x15.

Also, being sneaky should not be about exploiting the tiny view range (as if it was Deus Ex), but about real tactics and timing.

**Comment by Fox-McCloud** [comment]
updated version of this PR: https://github.com/ParadiseSS13/Paradise/pull/1260
